src/common_functions/mod.rs

```diff
@@ -0,0 +1,72 @@
+use bio::stats::probs::{LogProb, PHREDProb};
+use bio_types::sequence::SequenceRead;
+use ordered_float::NotNaN;
+use std::cmp;
+
+const PROB_CONFUSION: LogProb = LogProb(-1.0986122886681098); // (1 / 3).ln()
+const ALLELES: &'static [u8] = b"ACGT";
+
+pub trait CalcConsensus<'a, R: SequenceRead> {
+    fn validate_read_lengths(recs: &[R]) -> bool {
+        let reference_length = recs[0].len();
+        recs.iter()
+            .map(|rec| rec.len())
+            .all(|len| len == reference_length)
+    }
+
+    #[allow(unused_doc_comments)]
```

Comment:
Are you sure this attribute is necessary? I think traits may have doc comments.

Reply:
You are right. That was left over from the previous separate implementation.